testing/argsfilter/test_argsfilter_c.cpp

```diff
@@ -0,0 +1,395 @@
+ */
+TEST_P(argsfilter_c_p, socket_id) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+    char socket_id[2];
+
+    sprintf(socket_id, "0x%x", platform_.devices[0].socket_id);
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-S");
+    strcpy(three, socket_id);
+    EXPECT_EQ(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       socket_neg
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with invalid argument for the socket, <br>
+ *             the function returns an error. <br>
+ */
+TEST_P(argsfilter_c_p, socket_neg) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-S");
+    strcpy(three, "zxyw");
+    EXPECT_NE(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       segment
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with valid argument for the segment, <br>
+ *             the function returns 0. <br>
+ */
+TEST_P(argsfilter_c_p, segment) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+    char segment[2];
+
+    sprintf(segment, "0x%x", platform_.devices[0].segment);
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-s");
+    strcpy(three, segment);
+    EXPECT_EQ(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       argsfilter_neg
+ * @brief      Test: set_properties_from_args
+ * @details    When missing an argument for the bus, <br>
+ *             the function returns an error. <br>
+ */
+TEST_P(argsfilter_c_p, argsfilter_neg) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char *argv[] = { zero, one, two };
+    int argc = 3;
+    fpga_result result;
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-B");
+    EXPECT_NE(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+INSTANTIATE_TEST_CASE_P(argsfilter_c, argsfilter_c_p,
```

Comment:
add mock also

Reply:
These tests shouldn't need a sysfs (or mock sysfs) since they are only concerned with parsing arguments and filling in properties structure. Can we remove the dependency on the test_system and platform in the SetUp?